File: 8kyu/is-it-a-palindrome.js
//Describtion:
// Write a function that checks if a given string (case insensitive) is a
// palindrome. A palindrome is a word, number, phrase, or other sequence of 
// symbols that reads the same backwards as forwards, such as madam or 
// racecar, the date and time 12/21/33 12:21, and the
// sentence: "A man, a plan, a canal – Panama".

//My Solution
function isPalindrome(x) {    
    let input = x.toLowerCase()
    let y = input
    let p = input.split('').reverse().join('')
    if(p === y){
      return true
    }else{
      return false
    }
  }

//Other Solutions:
//Tenary operator
const isPalindrome = (x) => {
    return x.split("").reverse().join("").toLowerCase() === x.toLowerCase() ? true : false
  }

//Another Solution:
function isPalindrome(x, str = x.toLowerCase()) {
    return str === [...str].reverse().join('')
  }

File: Python/8kyu/is-it-a-palindrome.py
#Describtion:
# Write a function that checks if a given string (case insensitive) is a palindrome.

# A palindrome is a word, number, phrase, or other sequence of symbols that reads the same backwards as forwards, such as madam or racecar.

#Solution:
def is_palindrome(s):
    return True if s[::-1].lower() == s.lower() else False

#Other solutions:
def is_palindrome(s):
    s = s.lower()
    return s == s[::-1]
